sqlglot/dialects/singlestore.py

```diff
@@ -290,6 +290,24 @@ class Parser(MySQL.Parser):
             "USERS FOR GROUP": _show_parser("USERS FOR GROUP", target=True),
         }
 
+        def _parse_vector_expressions(self, expressions):
+            type_name = expressions[1].name
+            if type_name == "I8":
+                type_name = "TINYINT"
+            elif type_name == "I16":
+                type_name = "SMALLINT"
+            elif type_name == "I32":
+                type_name = "INT"
+            elif type_name == "I64":
+                type_name = "BIGINT"
+            elif type_name == "F32":
+                type_name = "FLOAT"
+            elif type_name == "F64":
+                type_name = "DOUBLE"
+
+            expressions[1] = expressions[0]
+            expressions[0] = exp.DataType.build(type_name, dialect=self.dialect)
```

Comment:
I don't like the fact that we're mutating the argument. Can we just build the data type and return it, instead?